preferences/preferencesdialog.cpp

```diff
@@ -0,0 +1,74 @@
+#include <QColorDialog>
+
+#include "preferencesdialog.h"
+#include "ui_preferencesdialog.h"
+#include "preferences.h"
+
+
+PreferencesDialog::PreferencesDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::PreferencesDialog)
+{
+    ui->setupUi(this);
+
+    connect(ui->listWidget, &QListWidget::currentRowChanged,
+            ui->stackedWidget, &QStackedWidget::setCurrentIndex);
+    // There are 2 signals - int and QString - we cannot enforce
+    // int one with eg. qOverload() as it's too new. So let's
+    // use old style for signals
+    //connect(ui->alphaSpinBox, &QSpinBox::valueChanged,
+    //        this, &PreferencesDialog::alphaSpinBox_valueChanged);
+    connect(ui->alphaSpinBox, SIGNAL(valueChanged(int)),
+            this, SLOT(alphaSpinBox_valueChanged(int)));
+
+    ui->listWidget->setCurrentRow(0);
+
+    m_colorMap[ui->hugeDiffPushButton] = &P->hugeDiffColor;
+    m_colorMap[ui->identicalDiffPushButton] = &P->identicalDiffColor;
+    m_colorMap[ui->moderateDiffPushButton] = &P->moderateDiffColor;
+    m_colorMap[ui->notPresentDiffButton] = &P->notPresentDiffColor;
+    m_colorMap[ui->keyWordPushButton] = &P->syntaxKeywordColor;
+    m_colorMap[ui->jsonKeysPushButton] = &P->syntaxValueColor;
+
+    QMap<QPushButton*, QColor*>::iterator i = m_colorMap.begin();
+    while (i != m_colorMap.end()) {
+        setupButton(i.key(), *i.value());
+        connect(i.key(), &QPushButton::clicked,
+                this, &PreferencesDialog::openColorDialog);
+        ++i;
+    }
+
+    ui->alphaSpinBox->setValue(P->diffColorsAlpha);
+}
+
+PreferencesDialog::~PreferencesDialog()
+{
+    delete ui;
+}
+
+void PreferencesDialog::setupButton(QPushButton *b, const QColor &c)
+{
+    QPixmap pm(64, 64);
+    pm.fill(c);
+    QIcon icon(pm);
+    b->setIcon(icon);
```

Comment:
```suggestion
    b->setIcon(icon);
    b->setStyleSheet("text-align:left;");
```

Reply:
In such case window looks more in order not like a colored chess board :)
Sure maybe you have another ideas how to align them better

Reply:
Text align QSS won't work in RTL language setups. What about to change the dialog to something like this?

![Screenshot_20201219_143724](https://user-images.githubusercontent.com/207237/102690752-e048be00-4207-11eb-8bc7-030150300c4a.png)

Reply:
I think it's perfect :)